model/constants.py

```diff
@@ -0,0 +1,260 @@
+    # wrangler] < [library staff] < [manual intervention]
+    #
+    # This list keeps track of the high-priority portion of that
+    # ordering.
+    #
+    # "LIBRARY_STAFF" comes from the Admin Interface.
+    # "MANUAL" is not currently used, but will give the option of putting in
+    # software engineer-created system overrides.
+    PRESENTATION_EDITION_PRIORITY = [METADATA_WRANGLER, LIBRARY_STAFF, MANUAL]
+
+class EditionConstants(object):
+    ALL_MEDIUM = object()
+    BOOK_MEDIUM = u"Book"
+    PERIODICAL_MEDIUM = u"Periodical"
+    AUDIO_MEDIUM = u"Audio"
+    MUSIC_MEDIUM = u"Music"
+    VIDEO_MEDIUM = u"Video"
+    IMAGE_MEDIUM = u"Image"
+    COURSEWARE_MEDIUM = u"Courseware"
+
+    ELECTRONIC_FORMAT = u"Electronic"
+    CODEX_FORMAT = u"Codex"
+
+    # These are the media types currently fulfillable by the default
+    # client.
+    FULFILLABLE_MEDIA = [BOOK_MEDIUM, AUDIO_MEDIUM]
+
+    medium_to_additional_type = {
+        BOOK_MEDIUM : u"http://schema.org/EBook",
+        AUDIO_MEDIUM : u"http://bib.schema.org/Audiobook",
+        PERIODICAL_MEDIUM : u"http://schema.org/PublicationIssue",
+        MUSIC_MEDIUM :  u"http://schema.org/MusicRecording",
+        VIDEO_MEDIUM :  u"http://schema.org/VideoObject",
+        IMAGE_MEDIUM: u"http://schema.org/ImageObject",
+        COURSEWARE_MEDIUM: u"http://schema.org/Course"
+    }
+
+    additional_type_to_medium = {}
+    for k, v in medium_to_additional_type.items():
+        additional_type_to_medium[v] = k
+
+class IdentifierConstants(object):
+    # Common types of identifiers.
+    OVERDRIVE_ID = u"Overdrive ID"
+    ODILO_ID = u"Odilo ID"
+    BIBLIOTHECA_ID = u"Bibliotheca ID"
+    GUTENBERG_ID = u"Gutenberg ID"
+    AXIS_360_ID = u"Axis 360 ID"
+    ELIB_ID = u"eLiburutegia ID"
+    ASIN = u"ASIN"
+    ISBN = u"ISBN"
+    NOVELIST_ID = u"NoveList ID"
+    OCLC_WORK = u"OCLC Work ID"
+    OCLC_NUMBER = u"OCLC Number"
+    # RBdigital uses ISBNs for ebooks and eaudio, and its own ids for magazines
+    RB_DIGITAL_ID = u"RBdigital ID"
+    OPEN_LIBRARY_ID = u"OLID"
+    BIBLIOCOMMONS_ID = u"Bibliocommons ID"
+    URI = u"URI"
+    DOI = u"DOI"
+    UPC = u"UPC"
+    BIBBLIO_CONTENT_ITEM_ID = u"Bibblio Content Item ID"
+    ENKI_ID = u"Enki ID"
+
+    DEPRECATED_NAMES = {
+        u"3M ID" : BIBLIOTHECA_ID,
+        u"OneClick ID" : RB_DIGITAL_ID,
+    }
+    THREEM_ID = BIBLIOTHECA_ID
+    ONECLICK_ID = RB_DIGITAL_ID
+
+    LICENSE_PROVIDING_IDENTIFIER_TYPES = [
+        BIBLIOTHECA_ID, OVERDRIVE_ID, ODILO_ID, AXIS_360_ID,
+        GUTENBERG_ID, ELIB_ID
+    ]
+
+    URN_SCHEME_PREFIX = "urn:librarysimplified.org/terms/id/"
+    ISBN_URN_SCHEME_PREFIX = "urn:isbn:"
+    GUTENBERG_URN_SCHEME_PREFIX = "http://www.gutenberg.org/ebooks/"
+    GUTENBERG_URN_SCHEME_RE = re.compile(
+        GUTENBERG_URN_SCHEME_PREFIX + "([0-9]+)")
+    OTHER_URN_SCHEME_PREFIX = "urn:"
+
+    IDEAL_COVER_ASPECT_RATIO = 2.0/3
+    IDEAL_IMAGE_HEIGHT = 240
+    IDEAL_IMAGE_WIDTH = 160
```

Comment:
I'm not sure what uses these ones but I never noticed them when I was working on covers. Seems like a weird place for them...

Reply:
Yes, this is weird -- they're here because Identifier is where we find the best cover image for a given identifier. This can also be resolved in a 'fix the constants' branch.